### PR TITLE
Adding options to plugin upload for Cloudinary.

### PR DESCRIPTION
**Pull request by Torone**
> ⚠️ We have stopped merging PRs for now to the Strapi core.<br><br>
> The reason is that we are developing new architecture for the admin panel and for the plugins.<br>
> This new architecture will provide stability of the Strapi core as we approach the release of Beta.<br>
> We appreciate and welcome all your contributions, but until further notice, please do not submit a PR as it will not be merged.<br>
> Furthermore, you will have to rewrite it based on the new architecture.

#### Description:
Cloudinary file upload is missing configuration options. This small change allow the user to easily create an option configuration that would be executed by Cloudinary:

```
const options = {
  transformation: { width: 800, height: 800, crop: "limit" },
  format: "jpg",
  quality: 80,
  folder: "my_folder"
};
```

Actually this is available just by hard coding the options, in future it would be great to move this into a configuration file or maybe into model.

#### My PR is a:
- [ ] 💥 Breaking change
- [ ] 🐛 Bug fix #issueNumber
- [x] 💅 Enhancement
- [ ] 🚀 New feature

#### Main update on the:
- [ ] Admin
- [ ] Documentation
- [ ] Framework
- [x] Plugin

#### Manual testing done on the following databases:
- [ ] Not applicable
- [x] MongoDB
- [ ] MySQL
- [ ] Postgres
- [ ] SQLite

### CODE REVIEW

File: packages/strapi-plugin-upload/services/Upload.js [hunk 1]
```diff
@@ -77,7 +77,11 @@ module.exports = {
     // Execute upload function of the provider for all files.
     return Promise.all(
       files.map(async file => {
-        await actions.upload(file);
+        
+        // This should host the provider configuration
+        const options = {};
```
Comment (alexandrebodin) [inline packages/strapi-plugin-upload/services/Upload.js]: The options are always an empty object. how should that work ?

Comment (Torone) [reply packages/strapi-plugin-upload/services/Upload.js]: By filling it as explained in the PR description.

Comment (alexandrebodin) [reply packages/strapi-plugin-upload/services/Upload.js]: This should come from the configuration not a codebase modification

Comment (Torone) [reply packages/strapi-plugin-upload/services/Upload.js]: Can you please point at this configuration so maybe I can add it in there?

**Comment by alexandrebodin** [comment]
Hi,

If you look at https://github.com/strapi/strapi/blob/master/packages/strapi-provider-upload-aws-s3/lib/index.js

You will see there is a `auth` field which can be used for any config info (we need to rename this field). You can see delcare a config attribute and  it will be accessible in the `init` function .
You can add the possible ressourceTypes like: 

```js
module.exports = {
  //...
  auth: {
    ressourceType: {
      label: 'resourceType',
      type: 'enmu',
      values: [
        'auto',
        // the other possible values 
      ]
    },
  },
};
```

**Comment by alexandrebodin** [comment]
Hi, I'm closing this issue as it's been inactive for a month. Feel free to reopen and update the PR ;)

**Pull request closed without merging**